### PR TITLE
backend/config: update infura API keys for clients with old URLs

### PR DESCRIPTION
**Pull request by Tomasvrba**

### CODE REVIEW

File: backend/config/config.go [hunk 1]
```diff
@@ -217,6 +218,7 @@ func NewConfig(appConfigFilename string, accountsConfigFilename string) *Config
 		accountsConfig:         newDefaultAccountsonfig(),
 	}
 	config.load()
+	config.appConfig, _ = config.checkAPIKeys(config.appConfig)
```
Comment (benma) [inline backend/config/config.go]: never ignore errors (rare exception when handling it is impossible or you know for a fact the error can't happen, but then it should be turned into a panic)

Comment (Tomasvrba) [reply backend/config/config.go]: Good catch, fixed.

File: backend/config/config.go [hunk 2]
```diff
@@ -217,7 +221,12 @@ func NewConfig(appConfigFilename string, accountsConfigFilename string) *Config
 		accountsConfig:         newDefaultAccountsonfig(),
 	}
 	config.load()
-	return config
+	var err error
+	config.appConfig, err = config.checkAPIKeys(config.appConfig)
```
Comment (benma) [inline backend/config/config.go]: Code reuse: use `config.SetAppConfig()` which sets and saves `config.appConfig.`

Naming: `checkAPIKeys` does not check, but modify the config. please use `updateInfuraAPIKeys()` or similar.

Comment (Tomasvrba) [reply backend/config/config.go]: Well, it checks and then modifies if the check fails :)
But you're right, `update` works better

Comment (Tomasvrba) [reply backend/config/config.go]: Updated.

File: backend/config/config.go [hunk 3]
```diff
@@ -217,7 +221,12 @@ func NewConfig(appConfigFilename string, accountsConfigFilename string) *Config
 		accountsConfig:         newDefaultAccountsonfig(),
 	}
 	config.load()
-	return config
+	appConfig, err := config.updateInfuraAPIKeys(config.appConfig)
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	config.SetAppConfig(appConfig)
```
Comment (benma) [inline backend/config/config.go]: handle error

File: backend/config/config.go [hunk 4]
```diff
@@ -270,3 +279,15 @@ func (config *Config) save(filename string, conf interface{}) error {
...
 	return errp.WithStack(ioutil.WriteFile(filename, jsonBytes, 0644))
 }
+
+func (config *Config) updateInfuraAPIKeys(appConfig AppConfig) (AppConfig, error) {
+	for _, ethConfig := range []*ethCoinConfig{&appConfig.Backend.ETH, &appConfig.Backend.TETH, &appConfig.Backend.RETH} {
+		if strings.HasSuffix(ethConfig.NodeURL, "infura.io/") {
+			ethConfig.NodeURL += infuraAPIKey
+		}
+	}
+	if err := config.save(config.appConfigFilename, appConfig); err != nil {
```
Comment (benma) [inline backend/config/config.go]: not needed, saving is already done in SetAppConfig()

**Pull request merged**